### DISCUSSION TITLE
Floswise Template for ready to use cases.

### DISCUSSION THREAD

**Discussion by akriot** [FlowiseAI/Flowise#16] [Ideas]
I am thinking to create a library of ready to use template for  common tasks. Let's discuss how useful this can be

**Comment by HenryHengZJ** [comment]
Are you thinking of something similar to Marketplace?
![screely-1681588366998](https://user-images.githubusercontent.com/26460777/232250591-2ad34e87-0dc3-44a8-8511-5bb220c5663e.png)

We'd really like to learn how people are using it! 
If you have use cases that you would like to make it into templates, you can do so by:
1.) Export your chatflow as json file
2.) Add the json file to `server/marketplaces`
3.) Add description to the json object. Example [here](https://github.com/FlowiseAI/Flowise/blob/main/packages/server/marketplaces/Antonym.json)
4.) Submit a PR

Reach out to us anytime here or [Discord](discord.gg/7C5xTWP8), happy to assist :)

**Reply by HenryHengZJ** [reply]
Also would be cool to make it into something similar to llama-hub where you have users who uploaded it, star count, and etc: ![llama-hub](https://camo.githubusercontent.com/c629d70de277acdd07526b4968cdc9082bfe52d38d4b634848a2e7c6e05c2534/68747470733a2f2f7363726162626c652d64696374696f6e6172792e73332e75732d776573742d322e616d617a6f6e6177732e636f6d2f53637265656e2b53686f742b323032332d30322d31312b61742b31322e34352e34342b504d2e706e67)

**Reply by akriot** [reply]
Yes, absolutely. We can have separate section for template gallery where user generated ChatFlow can be re-used for various use cases.This will look like marketplace but for ready to use templates. For example, one can go and select LinkedIn Template to start interacting with LinkedIn services. I have just started to play with this and getting some ideas.

**Reply by HenryHengZJ** [reply]
The templates in marketplace are ready to be used. Clicking `Use Template` will bring you to a new canvas with the same flow shown in the template, you just need to configure credentials like API key:

![screely-1681593943185](https://user-images.githubusercontent.com/26460777/232253775-50718afa-d8ec-4b1e-afa5-0da45cf04c62.png)

**Comment by iokinpardo** [comment]
This is awesome. Congratulations team.

**Question**: How can the final **chatbot be displayed on a web-site**?
Imagine we define a couple of those cases and would need to show them **online** inside our company **websites**. How could this be done?

**Reply by HenryHengZJ** [reply]
Very good question! And it has been asked lots of time too. That's the goal of the project, to be able to embed in your website application. We are working towards that, stay tuned for more updates :)

**Comment by mallelavamshi** [comment]
Hi,Luckily i found  WebPageQnA template for a particular usecase for me, All the components of the flow are working fine except the vector store part. I could only use in memory vector store but couldn't use any commercial vector stores like pinecone,milvus, Astra etc.. i tried 5 vector stores but no use, the embeddings are not getting upserted. only in memory vector store is working but that is not persistent for my use case. please help me how to solve this issue.

**Reply by toi500** [reply]
What error are u getting?

**Reply by mallelavamshi** [reply]
it is not showing any error message, simply error with in a redbox that's it

**Reply by toi500** [reply]
I would like to help you but I need to see a capture of your flow to check ur configuration, nodes, embedding nodes, etc.

Depending if you are using OpenAI, Gemini, Mistral etc you need to set up the vector store dimensions accordingly.

**Reply by mallelavamshi** [reply]
Thank you please have a look at the screen shots. The i have tried different vector stores with correct configurations like api keys and index names but none is working. Only in memory vector store is working. All i tried different webscrapers and embedding models but the problem seems to be only with vector store. here is the attached screen shots , i took pinecone here as an example.( i had setup the dimensions, index name and api key correctly)
![1](https://github.com/FlowiseAI/Flowise/assets/92425635/214565f2-def2-484e-890c-768f9cc695b2)
![2](https://github.com/FlowiseAI/Flowise/assets/92425635/78a11022-d366-4870-890b-cfc42e68f97d)
![3](https://github.com/FlowiseAI/Flowise/assets/92425635/ab5d761c-94a2-48b6-8a49-c02d2b0d1773)
![4](https://github.com/FlowiseAI/Flowise/assets/92425635/7a2272b4-d49b-4872-8bc2-9f733ae001a4)

**Comment by toi500** [comment]
I think that there are 2 problems here:

1. The Cohere `embed-english-v2.0` uses 4026 dimension-size, whitch is ridicully high for embedding simple text. Probabily it is not even compatible with a lot of vector stores.
2. You are no using an embedding from OpenAI. Since you are using a OpenAI node with `GPT 3.5 Turbo` as your chat model, you would need to use a OpenAI embedding to make it fully compatible.

Use the `text-embedding-ada-002` from OpenAI and set the dimension to `1536` at Pinecone.

--
![Screenshot 2024-05-03 010606](https://github.com/FlowiseAI/Flowise/assets/138339291/f82faa13-a6e8-4a9c-8574-cad2b3683124)
![Screenshot 2024-05-03 005808](https://github.com/FlowiseAI/Flowise/assets/138339291/10ea3d25-f28c-4309-968f-308356ba3357)
![Screenshot 2024-05-03 010540](https://github.com/FlowiseAI/Flowise/assets/138339291/cd8733bc-bb8f-4f77-8961-50dfa90bbb04)
<img width="1697" alt="image" src="https://github.com/FlowiseAI/Flowise/assets/138339291/256c7439-687f-4121-8841-2074e113eaf9">


It worked for me. Here is the flow I used:
[CRQAC Chatflow.json](https://github.com/FlowiseAI/Flowise/files/15194922/CRQAC.Chatflow.json)

**Reply by mallelavamshi** [reply]
Thank you very much for taking time to test the flow. I tried it  and surprisingly it worked. i was assuming the problem is with vector store components since in memory vector store is working fine.

**Comment by toi500** [comment]
The general rule is, if possible, to use the embeddings from the same provider that the chat model, to avoid unexpected problems.

Happy this solved for you ;)
